ocp_resources/resource.py

```diff
@@ -868,6 +868,11 @@ def instance(self):
         def _instance():
             return self.api.get(name=self.name)
 
+        if self.yaml_file and not self.name:
+            self.to_dict()
+            if not self.name:
+                raise ValueError(f"Resource name is {self.name}")
```

Comment:
under `if not self.name`; `self.name` can be dropped from the exception message

Reply:
If name gets set to a non-null value, but is still boolean false, knowing the value may be helpful